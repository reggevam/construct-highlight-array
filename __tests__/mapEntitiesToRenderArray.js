const mapEntitiesToRenderArray = require('./../src');
const smallText = require('../dummy-data/raw-text-small');
const smallLabels = require('../dummy-data/labels-small.js');

const largeText = require('../dummy-data/raw-text');
const largeLabels = require('../dummy-data/labels.js');

describe('mapEntitiesToRenderArray', () => {
  test('adds empty label tags to from raw text to highligh array on small version', () => {
    const result = mapEntitiesToRenderArray(smallText, smallLabels);
    expect(result).toMatchSnapshot();
  });

  test('adds empty label tags to from raw text to highligh array on large version', () => {
    const result = mapEntitiesToRenderArray(largeText, largeLabels);
    expect(result).toMatchSnapshot();
  });

  test('expects each odd entity to be labeled and each even entity to be unlabeled', () => {
    const result = mapEntitiesToRenderArray(largeText, largeLabels);

    const labeled = result.filter((_, ii) => ii % 2);
    const unlabeled = result.filter((_, ii) => ii % 2 === 0);

    labeled.forEach(entity => {
      expect(entity.isEntity).toBeTruthy();
    });

    unlabeled.forEach(entity => {
      expect(entity.isEntity).not.toBeTruthy();
    });
  });
});
