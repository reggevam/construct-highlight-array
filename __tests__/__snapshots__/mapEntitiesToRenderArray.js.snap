// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapEntitiesToRenderArray adds empty label tags to from raw text to highligh array on large version 1`] = `
Array [
  Object {
    "isEntity": false,
    "text": "The following is a statement of Patrolman ",
  },
  Object {
    "confidence": 0.5516320466995239,
    "endOffset": 52,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 42,
    "text": "Mark Coral",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": " with reference to the investigation of the homicide of ",
  },
  Object {
    "confidence": 0.9943994283676147,
    "endOffset": 117,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 108,
    "text": "Ana Smith",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": " age 31, married to ",
  },
  Object {
    "confidence": 0.9974563121795654,
    "endOffset": 147,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 137,
    "text": "Mark Smith",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": ".
On the morning of Sunday ",
  },
  Object {
    "confidence": 1,
    "endOffset": 184,
    "engine": "RegexEngine",
    "isEntity": true,
    "startOffset": 174,
    "text": "01/01/1970",
    "type": "DATE",
  },
  Object {
    "isEntity": false,
    "text": ", I was dispatched to 1600 ",
  },
  Object {
    "confidence": 0.4700889190038045,
    "endOffset": 240,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 211,
    "text": "Pennsylvania Avenue Northwest",
    "type": "LOC",
  },
  Object {
    "isEntity": false,
    "text": ", ",
  },
  Object {
    "confidence": 0.9996297359466553,
    "endOffset": 252,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 242,
    "text": "Washington",
    "type": "LOC",
  },
  Object {
    "isEntity": false,
    "text": ", ",
  },
  Object {
    "confidence": 0.6852731704711914,
    "endOffset": 256,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 254,
    "text": "DC",
    "type": "ORG",
  },
  Object {
    "isEntity": false,
    "text": ", ",
  },
  Object {
    "confidence": 0.9446771144866943,
    "endOffset": 261,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 258,
    "text": "USA",
    "type": "LOC",
  },
  Object {
    "isEntity": false,
    "text": " due to a call regarding a dead body.
Upon arrival of officer ",
  },
  Object {
    "confidence": 0.9982760548591614,
    "endOffset": 337,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 323,
    "text": "Margaret Heinz",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": " and myself, we met with ",
  },
  Object {
    "confidence": 0.9998801946640015,
    "endOffset": 372,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 362,
    "text": "Mark Smith",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": " the husband of the deceased. ",
  },
  Object {
    "confidence": 0.9995514154434204,
    "endOffset": 412,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 402,
    "text": "Mark Smith",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": " met us at the entrance to the 4 story building. There is no elevator in the building, one stair-way and no external fire escape. 
Before entering the building I noticed a city street camera right in front of the building.
The apartment is located on the second floor. Before entering, I asked Mr.",
  },
  Object {
    "confidence": 0.8572713732719421,
    "endOffset": 714,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 709,
    "text": " Smit",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": "h to tell me what had happened. Mr.",
  },
  Object {
    "confidence": 0.9910590052604675,
    "endOffset": 754,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 749,
    "text": " Smit",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": "h was very anxious and distressed. 
According to him, he left his residence last night at arou",
  },
  Object {
    "confidence": 1,
    "endOffset": 855,
    "engine": "RegexEngine",
    "isEntity": true,
    "startOffset": 848,
    "text": "nd 20:0",
    "type": "TIME",
  },
  Object {
    "isEntity": false,
    "text": "0 for his night shift and returned ",
  },
  Object {
    "confidence": 1,
    "endOffset": 896,
    "engine": "RegexEngine",
    "isEntity": true,
    "startOffset": 890,
    "text": "at 08:",
    "type": "TIME",
  },
  Object {
    "isEntity": false,
    "text": "00. He works as an operations dispatcher for th",
  },
  Object {
    "confidence": 0.8820345401763916,
    "endOffset": 964,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 943,
    "text": "e GAR ï¿½ Garbage Compa",
    "type": "ORG",
  },
  Object {
    "isEntity": false,
    "text": "ny. He said he was surprised to find the door unlocked and slightly opened as his wife was supposed to be at work already. He entered the apartment and called for his wife. As he entered the living room he foun",
  },
  Object {
    "confidence": 0.581649124622345,
    "endOffset": 1177,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 1174,
    "text": "d A",
    "type": "LOC",
  },
  Object {
    "isEntity": false,
    "text": "na on the floor with a head injury and a pool of blood around her. He ran out immediately to call the police. I asked Mr",
  },
  Object {
    "confidence": 0.9325100779533386,
    "endOffset": 1302,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 1297,
    "text": ". Smi",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": "th if he had touched the body or anything else and he said he did not. 
M",
  },
  Object {
    "confidence": 0.9919689297676086,
    "endOffset": 1380,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 1375,
    "text": "r. Sm",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": "ith's behavior seemed odd. I asked him when was the last time he was in contact with his wife and he replied it was the previous evening when he left for work. He tried calling her",
  },
  Object {
    "confidence": 1,
    "endOffset": 1567,
    "engine": "RegexEngine",
    "isEntity": true,
    "startOffset": 1560,
    "text": " at 06:",
    "type": "TIME",
  },
  Object {
    "isEntity": false,
    "text": "00 but she didn't answer.
No apparent forced entry was noticed. The body was on the floor with a head injury and a gun beside it. The apartment was very clean and neat. No note was found in the apartmen",
  },
  Object {
    "confidence": 0.4417806565761566,
    "endOffset": 1772,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 1769,
    "text": "t. ",
    "type": "LOC",
  },
  Object {
    "isEntity": false,
    "text": "Ana was fully dressed and our initial examination of the scene did not show any signs of burglary or assault. According to M",
  },
  Object {
    "confidence": 0.9899030923843384,
    "endOffset": 1901,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 1896,
    "text": "r. Sm",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": "it",
  },
  Object {
    "confidence": 0.4775758385658264,
    "endOffset": 1906,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 1903,
    "text": "h, ",
    "type": "LOC",
  },
  Object {
    "isEntity": false,
    "text": "Ana started her day",
  },
  Object {
    "confidence": 1,
    "endOffset": 1932,
    "engine": "RegexEngine",
    "isEntity": true,
    "startOffset": 1925,
    "text": " at 05:",
    "type": "TIME",
  },
  Object {
    "isEntity": false,
    "text": "30 every morning. She was a sales person ",
  },
  Object {
    "confidence": 0.9886243343353271,
    "endOffset": 1977,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 1973,
    "text": "at M",
    "type": "ORG",
  },
  Object {
    "isEntity": false,
    "text": "ACY's and has a two hour commute. 
A laptop and cellphone belonging to the deceased were filed into evidence.
 
The guns serial number foun",
  },
  Object {
    "confidence": 0.8539710640907288,
    "endOffset": 2123,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 2116,
    "text": "d was B",
    "type": "MISC",
  },
  Object {
    "isEntity": false,
    "text": "T17566
According to",
  },
  Object {
    "confidence": 0.996522068977356,
    "endOffset": 2147,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 2142,
    "text": " Mr. ",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": "Smith they do not own a gun.

I proceeded to question the neighbors living in the building.
The first floor neighbor i",
  },
  Object {
    "confidence": 0.9995540380477905,
    "endOffset": 2275,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 2265,
    "text": "s Mr. Jaco",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": "b Prat, 95 years old. He did not hear anything or see anything out of the ordinary. He was asl",
  },
  Object {
    "confidence": 1,
    "endOffset": 2376,
    "engine": "RegexEngine",
    "isEntity": true,
    "startOffset": 2369,
    "text": "eep at ",
    "type": "TIME",
  },
  Object {
    "isEntity": false,
    "text": "20:00 and woke",
  },
  Object {
    "confidence": 1,
    "endOffset": 2396,
    "engine": "RegexEngine",
    "isEntity": true,
    "startOffset": 2390,
    "text": " up at",
    "type": "TIME",
  },
  Object {
    "isEntity": false,
    "text": " 07:00.
The third floor neighbors ar",
  },
  Object {
    "confidence": 0.46701472997665405,
    "endOffset": 2443,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 2432,
    "text": "e the Swan ",
    "type": "LOC",
  },
  Object {
    "isEntity": false,
    "text": "Fa",
  },
  Object {
    "confidence": 0.998625636100769,
    "endOffset": 2449,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 2445,
    "text": "mily",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": ". Adam age 45 and his",
  },
  Object {
    "confidence": 0.9987843632698059,
    "endOffset": 2475,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 2470,
    "text": " wife",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": " Terry, they have a 5 year ol",
  },
  Object {
    "confidence": 0.6342423558235168,
    "endOffset": 2513,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 2504,
    "text": "d son ï¿½ P",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": "atrick that was at his grandmothers. Accordi",
  },
  Object {
    "confidence": 0.9955087900161743,
    "endOffset": 2561,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 2557,
    "text": "ng t",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": "o Ada",
  },
  Object {
    "confidence": 0.9599224925041199,
    "endOffset": 2571,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 2566,
    "text": "m and",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": " Terry they were  woken",
  },
  Object {
    "confidence": 1,
    "endOffset": 2601,
    "engine": "RegexEngine",
    "isEntity": true,
    "startOffset": 2594,
    "text": " up at ",
    "type": "TIME",
  },
  Object {
    "isEntity": false,
    "text": "05:30 by a loud bang that sounded like a car back fire. Following shortly they heard a car race off in the street adjacent to the building. They went back to sleep",
  },
  Object {
    "confidence": 0.6275870203971863,
    "endOffset": 2769,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 2764,
    "text": ".
The",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": " Swans had no real acquaintance wit",
  },
  Object {
    "confidence": 0.31907662749290466,
    "endOffset": 2810,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 2804,
    "text": "h the ",
    "type": "ORG",
  },
  Object {
    "isEntity": false,
    "text": "Smiths but did say that they were always shouting at each other.  
The fourth floor neig",
  },
  Object {
    "confidence": 0.9991068840026855,
    "endOffset": 2908,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 2898,
    "text": "hbor is Si",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": "mon Zeus a 29 year old student. He too was awoken by a loud noise but according to him the ",
  },
  Object {
    "confidence": 1,
    "endOffset": 3005,
    "engine": "RegexEngine",
    "isEntity": true,
    "startOffset": 2999,
    "text": "time w",
    "type": "TIME",
  },
  Object {
    "isEntity": false,
    "text": "as 04:30.",
  },
]
`;

exports[`mapEntitiesToRenderArray adds empty label tags to from raw text to highligh array on small version 1`] = `
Array [
  Object {
    "isEntity": false,
    "text": "The following is a statement of Patrolman ",
  },
  Object {
    "confidence": 0.5516320466995239,
    "endOffset": 52,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 42,
    "text": "Mark Coral",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": " with reference to the investigation of the homicide of ",
  },
  Object {
    "confidence": 0.9943994283676147,
    "endOffset": 117,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 108,
    "text": "Ana Smith",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": " age 31, married to ",
  },
  Object {
    "confidence": 0.9974563121795654,
    "endOffset": 147,
    "engine": "NeuralEngine",
    "isEntity": true,
    "startOffset": 137,
    "text": "Mark Smith",
    "type": "PER",
  },
  Object {
    "isEntity": false,
    "text": ".
On the morning of Sunday ",
  },
  Object {
    "confidence": 1,
    "endOffset": 184,
    "engine": "RegexEngine",
    "isEntity": true,
    "startOffset": 174,
    "text": "01/01/1970",
    "type": "DATE",
  },
  Object {
    "isEntity": false,
    "text": ", I was dispatched to 1600 Pennsylvania Avenue Northwest",
  },
]
`;
